### ISSUE TITLE
icetray-inspect -  on OSX `icetray.dylib` is included as an inspectable project (Trac #1604)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1596]
dumping `args` before `icetray-inspect` starts its magic:

```text
~/i3/combo/build
❯ ./env-shell.sh /Users/nega/i3/combo/build/bin/icetray-inspect --sphinx --sphinx-references --all --no-params --title="IceTracy Quick Reference" -o /Users/nega/i3/combo/build/sphinx_src/source/icetray_quick_reference.rst
['astro', 'bayesian_priors', 'CascadeVariables', 'clast', 'clsim', 'cmc', 'coinc_twc', 'CoincSuite', 'common_variables', 'common_variables__direct_hits', 'common_variables__hit_multiplicity', 'common_variables__hit_statistics', 'common_variables__track_characteristics', 'CommonVariables', 'coordinate_service', 'core_removal', 'corsika_reader', 'cramer_rao', 'credo', 'cscd_llh', 'daq_decode', 'dataclasses', 'dataio', 'DeepCore_Filter', 'diplopia', 'dipolefit', 'DOMLauncher', 'DomTools', 'double_muon', 'dst', 'earthmodel_service', 'examples_simulation', 'fill_ratio', 'filter_tools', 'filterscripts', 'finiteReco', 'frame_object_diff', 'full_event_followup', 'g4_tankresponse', 'gulliver', 'gulliver_modules', 'hdfwriter', 'HiveSplitter', 'i3math', 'IceHive', 'icepick', 'icetray', 'icetray.dylib', 'icetray_test', 'improvedLinefit', 'interfaces', 'ipdf', 'KalmanFilter', 'level3_filter_cascade', 'level3_filter_lowen', 'level3_filter_muon', 'lilliput', 'linefit', 'mue', 'MuonGun', 'neutrino_generator', 'NoiseEngine', 'paraboloid', 'payload_parsing', 'photonics_service', 'photospline', 'phys_services', 'portia', 'ppc', 'ppc_eff', 'ppc_pick', 'PROPOSAL', 'recclasses', 'rootwriter', 'shield', 'shovelart', 'shovelio', 'sim_services', 'simclasses', 'simprod', 'SLOPtools', 'smallshower_filter', 'spline_reco', 'static_twc', 'steamshovel', 'STTools', 'tableio', 'tensor_of_inertia', 'test_unregistered', 'topeventcleaning', 'TopologicalSplitter', 'toprec', 'topsimulator', 'tpx', 'trigger_sim', 'trigger_splitter', 'truncated_energy', 'VHESelfVeto', 'vuvuzela', 'WaveCalibrator', 'wavereform', 'weighting', 'wimpsim_reader', 'xppc']
```

`icetray-inspect` then proceeds to fail:

```text
icepick
Loading icepick...........................................ok
icetray
icetray.dylib
*** Failure loading 'libicetray.dylib'.
*** Load external libraries without an extension.
*** e.g. please omit '.dylib' or '.so'.

```

Migrated from https://code.icecube.wisc.edu/ticket/1604
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:42", 
    "description": "dumping `args` before `icetray-inspect` starts its magic:\n\n{{{\n~/i3/combo/build\n\u276f ./env-shell.sh /Users/nega/i3/combo/build/bin/icetray-inspect --sphinx --sphinx-references --all --no-params --title=\"IceTracy Quick Reference\" -o /Users/nega/i3/combo/build/sphinx_src/source/icetray_quick_reference.rst\n['astro', 'bayesian_priors', 'CascadeVariables', 'clast', 'clsim', 'cmc', 'coinc_twc', 'CoincSuite', 'common_variables', 'common_variables__direct_hits', 'common_variables__hit_multiplicity', 'common_variables__hit_statistics', 'common_variables__track_characteristics', 'CommonVariables', 'coordinate_service', 'core_removal', 'corsika_reader', 'cramer_rao', 'credo', 'cscd_llh', 'daq_decode', 'dataclasses', 'dataio', 'DeepCore_Filter', 'diplopia', 'dipolefit', 'DOMLauncher', 'DomTools', 'double_muon', 'dst', 'earthmodel_service', 'examples_simulation', 'fill_ratio', 'filter_tools', 'filterscripts', 'finiteReco', 'frame_object_diff', 'full_event_followup', 'g4_tankresponse', 'gulliver', 'gulliver_modules', 'hdfwriter', 'HiveSplitter', 'i3math', 'IceHive', 'icepick', 'icetray', 'icetray.dylib', 'icetray_test', 'improvedLinefit', 'interfaces', 'ipdf', 'KalmanFilter', 'level3_filter_cascade', 'level3_filter_lowen', 'level3_filter_muon', 'lilliput', 'linefit', 'mue', 'MuonGun', 'neutrino_generator', 'NoiseEngine', 'paraboloid', 'payload_parsing', 'photonics_service', 'photospline', 'phys_services', 'portia', 'ppc', 'ppc_eff', 'ppc_pick', 'PROPOSAL', 'recclasses', 'rootwriter', 'shield', 'shovelart', 'shovelio', 'sim_services', 'simclasses', 'simprod', 'SLOPtools', 'smallshower_filter', 'spline_reco', 'static_twc', 'steamshovel', 'STTools', 'tableio', 'tensor_of_inertia', 'test_unregistered', 'topeventcleaning', 'TopologicalSplitter', 'toprec', 'topsimulator', 'tpx', 'trigger_sim', 'trigger_splitter', 'truncated_energy', 'VHESelfVeto', 'vuvuzela', 'WaveCalibrator', 'wavereform', 'weighting', 'wimpsim_reader', 'xppc']\n}}}\n\n`icetray-inspect` then proceeds to fail:\n{{{\nicepick\nLoading icepick...........................................ok\nicetray\nicetray.dylib\n*** Failure loading 'libicetray.dylib'.\n*** Load external libraries without an extension.\n*** e.g. please omit '.dylib' or '.so'.\n\n}}}\n", 
    "reporter": "nega", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067102580394", 
    "component": "infrastructure", 
    "summary": "icetray-inspect -  on OSX `icetray.dylib` is included as an inspectable project", 
    "priority": "major", 
    "keywords": "documentation", 
    "time": "2016-03-28T17:34:57", 
    "milestone": "", 
    "owner": "kjmeagher", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160328T17:37:56: 
* nega changed component from "cmake" to "infrastructure"
* nega changed keywords from "" to "documentation"

**Comment by icecube-trac** [comment]
Trac update at 20160420T18:52:16: olivas changed milestone from "March 2016" to "April 2016"

**Comment by icecube-trac** [comment]
Trac update at 20160506T12:13:18: 
* kjmeagher changed resolution from "" to "fixed"
* kjmeagher commented:

> In changeset 2568:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2568"
> redo the way icetray-inspect handles available libs, cmake writes a zero length file to `${I3_BUILD}/docs/no_inspect/` for projects which can't be inspected and icetray-inspect checks there for projects to skip
> fixes #1596,#1597 and #1309
> ```


* kjmeagher changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T20:46:33: nwhitehorn commented:

> In changeset 168734:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168734"
> redo the way icetray-inspect handles available libs, cmake writes a zero length file to `${I3_BUILD}/docs/no_inspect/` for projects which can't be inspected and icetray-inspect checks there for projects to skip
> fixes #1596,#1597 and #1309
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:17:55: nwhitehorn commented:

> In changeset 170271:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170271"
> redo the way icetray-inspect handles available libs, cmake writes a zero length file to `${I3_BUILD}/docs/no_inspect/` for projects which can't be inspected and icetray-inspect checks there for projects to skip
> fixes #1596,#1597 and #1309
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:42: 
* olivas commented:

> Milestone April 2016 deleted


* olivas changed milestone from "April 2016" to ""
